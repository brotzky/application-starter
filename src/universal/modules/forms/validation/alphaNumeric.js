import validator from 'validator';

export default function(value) {
  return validator.isAlphanumeric(value)
    ? undefined
    : {
        defaultMessage: 'cannot contain special characters',
        id: 'form.errors.alphaNumeric',
        values: {},
      };
}
