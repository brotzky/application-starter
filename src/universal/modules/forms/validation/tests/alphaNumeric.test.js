import alphaNumeric from '../alphaNumeric';

describe('[Form Validation] alphaNumeric', () => {
  describe('alphaNumeric()', () => {
    it('should return no errors if a string of numbers is passed in', () => {
      expect(alphaNumeric('111')).toMatchSnapshot();
    });

    it('should return no errors if a string of only alphabetical letters is passed in', () => {
      expect(alphaNumeric('asdf')).toMatchSnapshot();
    });

    it('should return an error object when special characeters are passed in', () => {
      expect(alphaNumeric('&!(#')).toMatchSnapshot();
    });
  });
});
