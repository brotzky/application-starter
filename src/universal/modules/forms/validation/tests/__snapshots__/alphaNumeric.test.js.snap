// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Form Validation] alphaNumeric alphaNumeric() should return an error object when special characeters are passed in 1`] = `
Object {
  "defaultMessage": "cannot contain special characters",
  "id": "form.errors.alphaNumeric",
  "values": Object {},
}
`;

exports[`[Form Validation] alphaNumeric alphaNumeric() should return no errors if a string of numbers is passed in 1`] = `undefined`;

exports[`[Form Validation] alphaNumeric alphaNumeric() should return no errors if a string of only alphabetical letters is passed in 1`] = `undefined`;
